100.extend-exercise/convert-a-binary-number-to-decimal.rb

```diff
@@ -0,0 +1,26 @@
+require 'byebug'
+class ConvertABinaryNumberToDecimal
+  def self._2dec(number)
+    return 0 if number == 0
+    switch(number)
+  end
+
+  def self.switch(number)
+    initial_arrays = number.to_s.split("")
+    arrays = []
+    result = []
+    length = number.to_s.split("").length - 1
+    (0..length).to_a.each do |element|
+      arrays << 2**element
+    end
+    result << initial_arrays
+    result << arrays.reverse
+    b = []
+    result.transpose.each do |element|
+      b << element[0].to_i*element[1]
+    end
+    b.sum
+  end
```

Comment:
I can't get your algorithm here. Could you explain it? The code looks complicated.

Reply:
I think we can use `each_with_index` to simplify the code.
```
number.to_s.split("").reverse.each_with_index |digit, index|
  result += digit * (2**index)
end
```

Reply:
I've done 

https://github.com/dhoangk07/exercism-ruby/pull/19/commits/4b971f2961feae1f9ba447278aabc5130f711996